src/ImageSharp/IO/ChunkedMemoryStream.cs

```diff
@@ -12,44 +14,23 @@ namespace SixLabors.ImageSharp.IO;
 /// Chunks are allocated by the <see cref="MemoryAllocator"/> assigned via the constructor
 /// and is designed to take advantage of buffer pooling when available.
 /// </summary>
-internal sealed class ChunkedMemoryStream : Stream
+public class ChunkedMemoryStream : Stream
 {
-    // The memory allocator.
-    private readonly MemoryAllocator allocator;
+    private readonly MemoryChunkBuffer memoryChunkBuffer;
+    private readonly byte[] singleByteBuffer = new byte[1];
 
-    // Data
-    private MemoryChunk? memoryChunk;
-
-    // The total number of allocated chunks
-    private int chunkCount;
-
-    // The length of the largest contiguous buffer that can be handled by the allocator.
-    private readonly int allocatorCapacity;
-
-    // Has the stream been disposed.
+    private long length;
```

Comment:
Sidenote: IMO it's fairly conventional in C# land to prefix private field names with an underscore, to allow easy distinction from local variables, and to avoid excessive use of `this.`. I assume this is your personal preference, but thought I'd mention it as IMO it is somewhat non-idiomatic C#.

Reply:
Thanks, but I'd rather stick to using the language as designed rather than using conventions carried over from C. 

The `this` keyword provides important context IMO and encouraged consistancy throughout a codebase.

Reply:
@colgreen the project's preferred coding style is based on [Framework design guidelines](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/), and on StyleCop recommendations. The guidelines [explicitly prohibit](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-fields) prefixing variables.

Reply:
The guidelines state `Internal and private fields are not covered by guidelines,` (I believe those guidelines are primarily related to public API surface, rather than private/internal naming). The underscore prefix for private fields is very common in my experience, e.g. it's used widely in Microsoft .NET repos.

However, this topic is probably not relevant in the context of this PR :)